### ISSUE TITLE
Support for shutdown grace period in netty clients

### ISSUE THREAD

**Issue by dsyer** [spring-projects-experimental/spring-grpc#12]

**Comment by panic08** [comment]
Do I understand correctly that the purpose of this issue is to create a “life cycle” beans for a client? Which are already in place for the server

**Comment by dsyer** [comment]
No, I don't think so. The lifecycle is already handled by the `DefaultGrpcChannelFactory` (it is a `DisposableBean`). But there is no explicit configuration support for graceful shutdown (waiting for ongoing requests to finish before shutdown). I believe it is probably supported in the client, but we don't expose it anywhere.

**Comment by panic08** [comment]
> No, I don't think so. The lifecycle is already handled by the `DefaultGrpcChannelFactory` (it is a `DisposableBean`). But there is no explicit configuration support for graceful shutdown (waiting for ongoing requests to finish before shutdown). I believe it is probably supported in the client, but we don't expose it anywhere.

Yes, `DefaultGrpcChannelFactory` implements `DisosableBean` and overrides the destroy method which closes channels without any delay. I guess the point of the question is to allow the user to specify a delay(grace period) after which the channel shutdown?

Can I be assigned to this issue?
